deployments/puppet/manifests/collector_win_registry.pp

```diff
@@ -16,7 +16,10 @@
     "SPLUNK_REALM=${splunk_otel_collector::splunk_realm}",
     "SPLUNK_TRACE_URL=${splunk_otel_collector::splunk_trace_url}",
   ]
-  + if !$splunk_otel_collector::splunk_ballast_size_mib.strip().empty() {
+  + if ($splunk_otel_collector::params::collector_version != 'latest' or
+  versioncmp($splunk_otel_collector::params::collector_version, '0.97.0') < 0 ) and
+  !$splunk_otel_collector::splunk_ballast_size_mib.strip().empty()
```

Comment:
Do we need to worry about running a new puppet release with an old collector? Maybe just remove it and document that new puppet release requires 0.97.0+?

Reply:
Aren't we use latest for Windows as well? @jeffreyc-splunk WDYT?

Reply:
@dmitryax  FYI-> https://github.com/signalfx/splunk-otel-collector/blob/main/deployments/puppet/manifests/params.pp#L30

Reply:
We can make it "latest" but not sure why the default is an empty string.

Reply:
For windows, the `collector_version` parameter is required in the user's manifest and does not currently support `latest` (see https://github.com/signalfx/splunk-otel-collector/blob/main/deployments/puppet/README.md#collector).

Reply:
`GOMEMLIMIT` will need to be added to the windows registry if it is non-empty and `collector_version` >= 0.97.0.

Reply:
Ok, sounds good. We should keep it then. @jeffreyc-splunk is there a way to have a default collector version for Windows going forward?

Reply:
> Ok, sounds good. We should keep it then. @jeffreyc-splunk is there a way to have a default collector version for Windows going forward?

We could simply set the default version to something specific, e.g. `0.97.0`, but it may require us to bump this version and release the puppet module for each collector release.  And setting the default version for windows to `latest` would be more complex since there isn't a repo manager like yum or apt for MSI's that can handle determining the latest release.

Also, to help maintain idempotence and prevent inadvertent upgrades in production environments, it is recommended for the user to pin the desired version in their manifests.

Reply:
I think setting the default value to the latest release is fine if we have a release automation that would take care of this. Will the default values be applied to both Windows and Linux?

Reply:
per discussion offline, we have decided to have this change in a different scope. It is beyond this PR's scope for now.